src/semantic_world/world_entity.py

```diff
@@ -293,13 +284,13 @@ def bodies(self) -> Iterable[Body]:
         """
         return self._bodies(set())
 
-    def as_bounding_box_collection(self) -> BoundingBoxCollection:
+    def as_bounding_box_collection(self, reference_frame: Optional[Body] = None) -> BoundingBoxCollection:
         """
         Returns a bounding box collection that contains the bounding boxes of all bodies in this view.
         """
         bbs = reduce(
             lambda accumulator, bb_collection: accumulator.merge(bb_collection),
-            (body.bounding_box_collection for body in self.bodies if body.has_collision())
+            (body.as_bounding_box_collection(reference_frame) for body in self.bodies if body.has_collision())
```

Comment:
sorry, but this is a horribly complicated line of code. this should be a loop instead.